packages/mux-player-react/src/index.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect } from "react";
+import "@mux-elements/mux-player";
+import type MuxPlayerElement from "@mux-elements/mux-player";
+import { toNativeProps } from "./common/utils";
+import { useRef } from "react";
+import { useCombinedRefs } from "./useCombinedRefs";
+
+type ValueOf<T> = T[keyof T];
+
+export type MuxPlayerRefAttributes = MuxPlayerElement;
+
+type VideoApiAttributes = {
+  currentTime: number;
+  volume: number;
+  src: string | null;
+  poster: string;
+  playbackRate: number;
+  crossOrigin: string;
+  autoPlay: boolean;
+  loop: boolean;
+  muted: boolean;
+};
+type StreamTypes = {
+  VOD: "on-demand";
+  LIVE: "live";
+  LL_LIVE: "ll-live";
+};
+
+type MuxMediaPropTypes = {
+  // envKey: Options["data"]["env_key"];
+  envKey: string;
+  // debug: Options["debug"] & Hls["config"]["debug"];
+  debug: boolean;
+  // metadata: Partial<Options["data"]>;
+  metadata: { [k: string]: any };
+  beaconDomain: string;
+  playbackId: string;
+  preferMse: boolean;
+  streamType: ValueOf<StreamTypes> | "vod";
+  startTime: number;
+  children: never[];
+};
+
+export type MuxPlayerProps = {
+  forwardSeekOffset?: number;
+  backwardSeekOffset?: number;
+  metadataVideoId?: string;
+  metadataVideoTitle?: string;
+  metadataViewerUserId?: string;
```

Comment:
In React world, I think it makes sense to handle `metadata` like we did in `<MuxVideo />`:

```jsx
        metadata={{
          video_id: "video-id-123456",
          video_title: "Super Interesting Video",
          viewer_user_id: "user-id-bc-789",
        }}
```

Let me know if that makes sense

Reply:
Yup definitely does. We can let users let us know if they'd prefer to also have the shorthand and introduce it if they do.